source/common/router/upstream_request.cc

```diff
@@ -106,6 +106,18 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
       }
     }
   }
+
+  // The pool factory will ensure the conn pool is valid before return it to the router filter.
```

Comment:
I think we do this check in the router so how about
// The router checks that the connection pool is non-null before creating the upstream request.

Reply:
in the more deep impl detail, is the pool factory do the check. it ensure all the returned non-null pools will have valid pool data and host. But may be it too deep.

I will update the comment.